cms/djangoapps/contentstore/views/course.py

```diff
@@ -1069,11 +1034,20 @@ def _modify_tabs_to_components(request, course_module):
         'open_ended': (is_advanced_component_present, OPEN_ENDED_COMPONENT_TYPES),
         # notes tab
         'notes': (is_advanced_component_present, NOTE_COMPONENT_TYPES),
-        # student notes tab
-        'edxnotes': (is_field_value_true, ['edxnotes'])
     }
 
-    tabs_changed = False
+    def update_tab(tabs, tab_type, tab_enabled):
+        """
+        Adds or removes a course tab based upon whether it is enabled.
+        """
+        tab_panel = _get_tab_panel_for_type(tab_type)
+        if tab_enabled:
+            tabs.append(CourseTab.from_json(tab_panel))
+        elif tab_panel in tabs:
+            tabs.remove(tab_panel)
```

Comment:
Having this function mutate `tabs` seems like it's liable to lead to tab duplication down the line. It might be better to simply have a list of possible tabs, and then filter those into a clean list based on whether they're enabled or not.